sklearn/linear_model/_logistic.py

```diff
@@ -2094,7 +2106,14 @@ def fit(self, X, y, sample_weight=None):
 
         # init cross-validation generator
         cv = check_cv(self.cv, y, classifier=True)
-        folds = list(cv.split(X, y))
+        scorer = get_scorer(self.scoring)
+        cv_params = metadata_request_factory(cv).split.get_method_input(
+            ignore_extras=True, **fit_params
+        )
+        score_params = metadata_request_factory(scorer).split.get_method_input(
+            ignore_extras=True, **fit_params
+        )
+        folds = list(cv.split(X, y, **cv_params))
```

Comment:
This design doesn't seem to ensure that all `fit_params` have been requested by something, does it? I really want to be assured that typos get caught.

One syntax that would allow for this assurance is more like:
```
cv_params, score_params = prepare_requested_metadata((cv, 'split'), (scorer, 'split'))
```

Reply:
This is a very tricky case. Here `LogisticRegressionCV` is _kind of_ a CV object, but also a normal _consumer_ estimator. That means if the user passes `sample_weight` without requesting it explicitly, nothing should break. 

We could also say if the user wants to pass sample_weight, they have to say that explicitly as:

``` python
LogisticRegressionCV().fit_requests(sample_weight=True).fit(X, y, sample_weight=sample_weight)
```

But that diverges from our decision that _estimators_ shouldn't _validate_ the input. Also, what do we do with

``` python
LogisticRegressionCV().fit_requests(sample_weight="some_other_name").fit(X, y, sample_weight=sample_weight)
```

?

With the above in mind, I don't see any easy way to _validate_ the inputs in `LogisticRegressionCV`, as done in `Pipeline` or `GridSearchCV` for instance.

To be concrete, what do we do when `scorer` has sample weight set to `UNREQUESTED`, but the user passes `sample_weight`? Your construct would error cause `sample_weight` is not requested by scorer or the splitter, but it shouldn't.

Reply:
You're right that it needs another term with a request for its own usage: `self_params, cv_params, score_params = ...` where self always uses the prop that is here named `sample_weight` (does that make sense? I guess its request is invalid if sample_weight is unrequested by the estimator but requested by its scorer without aliasing!)

Reply:
I added a `self_metadata` to the `validate` method, to keep the change minimal. I kinda like this API better than the function you suggest since it keeps the logic of how the routing should work where it is (in `get_metadata_request`). WDYT?